cloud_chatops/src/get_github_prs.py

```diff
@@ -53,57 +85,28 @@ def get_http_response(self, url: str) -> List[Dict]:
         """
         headers = {"Authorization": "token " + get_token("GITHUB_TOKEN")}
         response = requests.get(url, headers=headers, timeout=60)
-        self.validate_response(response, url)
+        self._validate_response(response)
         return response.json()
 
-    def format_http_responses(
-        self, responses: Union[Dict[str, List], Dict[str, Dict]]
-    ) -> Dict[str, List]:
-        """
-        This method checks the formats the responses from GitHub are in a consistent format.
-        :param responses: GitHub's HTTP responses.
-        :return: Dictionary of responses.
-        """
-        culled_responses = self.remove_empty_response(responses)
-        for repo, response in culled_responses.items():
-            if isinstance(response, dict):
-                responses[repo] = [response]
-        return responses
-
     @staticmethod
-    def remove_empty_response(
-        responses: Union[Dict[str, List], Dict[str, Dict]]
-    ) -> Union[Dict[str, List], Dict[str, Dict]]:
-        """
-        This method removes all empty responses from the Dictionary.
-        An empty response is the result of no open pull requests.
-        :param responses: The responses to check.
-        :return: The responses with no empty values.
-        """
-        to_remove = []
-        for repo, response in responses.items():
-            if not response:
-                to_remove.append(repo)
-
-        for i in to_remove:
-            del responses[i]
-        return responses
-
-    @staticmethod
-    def validate_response(response: requests.get, url: str) -> None:
+    def _validate_response(response: requests.get) -> None:
         """
         This method checks the status code of the HTTP response and handles exceptions accordingly.
         :param response: The response to check.
-        :param url: The url that was not found.
         """
-        if response.status_code == 401:
-            raise BadGitHubToken(
-                "Your GitHub api token is invalid. Check that it hasn't expired."
-            )
-        if response.status_code == 404:
-            raise RepoNotFound(f'The repo at the url "{url}" could not be found.')
+        match response.status_code:
```

Comment:
nice to see usage of case statements :) - though its quite a new thing in Python

Since you're containerizing this script you can leave as is. But if you want it to run on older versions of python - you can use a dictionary like this:

```python
    { 
        200: lambda: pass
        401: lambda: raise BadGitHubToken(...),
        404: lambda: raise RepoNotFound(...)
    }.get(response.status_code, lambda: UnknownHTTPError(...))()
```

But I do agree the case statement is much cleaner

Reply:
That's useful to know, I haven't seen it before.